src/sync.cpp

```diff
@@ -1950,5 +1959,572 @@ bool Sync::movetolocaldebris(LocalPath& localpath)
 
     return false;
 }
+
+m_off_t Sync::getInflightProgress()
```

Comment:
Not sure how the previous implementation was done, but this one requires to iterate through all transfer slots for every sync, every 30s. Taking into account the loop only takes into account active transfers, it shouldn't be time-consuming, right?

Reply:
Yes - as it's once per 30 seconds it doesn't cost much - and we only ever have max 30 items (maybe more in the future but never an excessive amount) in the tslots list, so this is pretty trivial anyway.    The older version was based on a listener callback for every progress update of every transfer, all the time, and then looking that up in a map of active transfers, so probably considerably more costly.